site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,316 @@
+
+Clients use this command to initially *add* credentials for a device that has already been registered with Hono. The credentials to be added may be of arbitrary type. However, [Standard Credential Types]({{< relref "#standard-credential-types" >}}) contains an overview of some common types that are used by Hono's protocol adapters and which may be useful to others as well.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *add credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type            | Description                         |
+| :--------------- | :-------: | :----------------------- | :-------------- | :---------------------------------- |
+| *device_id*      | yes       | *application-properties* | UTF-8 *string*  | MUST contain the ID of the device.  |
+| *action*         | yes       | *application-properties* | UTF-8 *string*  | MUST contain the value `add`.       |
+
+The request message MUST include payload as defined in [Request Payload]({{< relref "#request-payload" >}}).
+
+**Response Message Format**
+
+A response to an *add credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty. 
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *201* | Created, the credentials have been added successfully. |
+| *409* | Conflict, there already exist credentials with the `type` and `auth-id` provided in the request payload for the *device_id*. |
+| *412* | Precondition Failed, there is no device registered with the given *device_id* within the *tenant_id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get Credentials
+
+Protocol adapters use this command to *look up* credentials of a particular type for a device identity.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on an *get credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type            | Description                   |
+| :--------------- | :-------: | :----------------------- | :-------------- | :---------------------------- |
+| *action*         | yes       | *application-properties* | UTF-8 *string*  | MUST contain the value `get`. |
+
+The body of the message MUST consist of a single *AMQP Value* section containing a UTF-8 encoded string representation of a single JSON object as follows:
+
```

Comment:
I was thinking also yesterday in general of using JSON in API instead of application properties (or the AMQP value). I think we can cover the most without including another encoding (JSON), but we use it on different places already, so we probably want to be consistent (even in this case).

I'd appreciate to hear more thoughts on this.